gui/src/components/RegisterCluster.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <div class="modal fade" tabindex="-1" id="registerCluster" aria-labelledby="registerClusterModal"
+         aria-hidden="true">
+        <div class="modal-dialog modal-dialog-centered">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title">Register a new cluster</h5>
+                    <button class="close" type="button" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <form class="needs-validation" novalidate>
+                        <div class="form-group">
+                            <label for="cassandraHostInput">Cassandra host address</label>
+                            <input class="form-control" type="text" id="cassandraHostInput" ref="cassandraHostInput"
+                                   placeholder="localhost" required>
+                            <div class="invalid-feedback">
+                                Please enter a host address.
+                            </div>
+                            <small v-if="failed" class="form-text text-danger">{{ error }}</small>
+                        </div>
+                        <div class="modal-footer">
+                            <button class="btn btn-primary" data-target="#registerCluster" type="submit"
+                                    @click="registerCluster">Register
+                            </button>
+                        </div>
+                    </form>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+  import $ from 'jquery';
+
+  export default {
+    data() {
+      return {
+        failed: false,
+        error: "",
+      }
+    },
+    methods: {
+      registerCluster() {
+        this.$api.post('/clusters', {
+          cassandra_host: document.getElementById('cassandraHostInput').value,
+        }).then((response) => {
+          if (response.status === 200) {
```

Comment:
If it doesn't return `200`, it doesn't happen anything ?
I think some error handling is needed.
The same for the other parts.

Reply:
I found that the way to catch errors like `400` or `500` etc, you need to add a `catch` block. So in this case and some of the other code, the `.catch(error)` handles cases when the status is not 200.

Reply:
However, there are some parts in the code where more robust error handling could be good. RegisterCluster currently catches the most common error responses, but could probably use a general `else { ... }` to catch unexpected issues.

Reply:
Not fully sure what you said but are you saying error handling should be generalized as a function and it should be used for other similar code ?
If so, I agree.
Anyways, I think we should do error handling; at least showing some error message to somewhere like at the bottom.

Error handling can be done in another PR .

Reply:
What I mean is this: The axios API seems to interpret codes such as `400` and `500` as errors. So, if you write a line such as `else if (response.status === 500)`, that code will never be reached and it will just throw an error to the console because the request failed.

Instead, you need to `catch` the `400` / `500` status codes like in the code below. This is the way that I've implemented error handling in all the classes so far.

```java
        this.$api.post('/clusters', {
          cassandra_host: document.getElementById('cassandraHostInput').value,
        }).then((response) => {
          if (response.status === 200) {
            this.failed = false;
            $('#registerCluster').modal('hide');
            this.$emit('updateClusterList');
          }
        })
        .catch(error => {                                    // here is where the error handling is done
          this.failed = true;
          if (error.response.data.code === 13) {
            this.error = "Failed to register cluster. Please make sure Cassandra is started."
          } else if (error.response.status === 503) {
            this.error = "Failed to register cluster. Cassy service unavailable."
          }
        })
        ;
```

Reply:
Oh, OK. So error handling is done but errors are only shown in console logs.
I think it will be more user friendly if we have some error message space at around the bottom. 
(or at the top. I'm not sure where to put honestly, but there must be some standard in this kind of thing)

@Torch3333  What do you think ?

Reply:
Not quite. What I mean is, Axios has a default way of handling errors (it sends console logs). However, I am catching that error to tell it what to do. In the example above, I'm setting a data variable called `error` to a message pertaining to what caused the error.

This message is displayed at the bottom of the RegisterCluster modal, like this:
<img width="496" alt="Screen Shot 2020-07-03 at 10 11 14" src="https://user-images.githubusercontent.com/46988771/86421623-afd29d80-bd15-11ea-9d9a-5ec19d963852.png">